test/translate_c.zig

```diff
@@ -1553,6 +1553,40 @@ pub fn addCases(cases: *tests.TranslateCContext) void {
         "pub const NOT_ZERO = ~c_uint(0);",
     );
 
+    cases.addC("pointer conversion with different alignment",
+        \\void test_ptr_cast() {
+        \\    void *p;
+        \\    {
+        \\        char *to_char = (char *)p;
+        \\        short *to_short = (short *)p;
+        \\        int *to_int = (int *)p;
+        \\        long long *to_longlong = (long long *)p;
+        \\    }
+        \\    {
+        \\        char *to_char = p;
+        \\        short *to_short = p;
+        \\        int *to_int = p;
+        \\        long long *to_longlong = p;
+        \\    }
+        \\}
+        ,
+        \\pub export fn test_ptr_cast() void {
+        \\    var p: ?*c_void = undefined;
+        \\    {
+        \\        var to_char: [*c]u8 = @ptrCast([*c]u8, p);
+        \\        var to_short: [*c]c_short = @ptrCast([*c]c_short, @alignCast(2, p));
```

Comment:
maybe this is easier said than done but this would be better translated as `@alignCast(@alignOf(c_short), p)` rather than hard coding 2.